diff --git a/src/compiler/parse/acorn.js b/src/compiler/parse/acorn.js
index 85d219d8c9a024e0cb35e9554d8fef6c0458c3fd..2a104636d9801312f0099b3238e480976befe609 100644
--- a/src/compiler/parse/acorn.js
+++ b/src/compiler/parse/acorn.js
@@ -1,13 +1,13 @@
 import * as code_red from 'code-red';
+import babel from '@babel/parser';
 
 /**
  * @param {string} source
  */
 export const parse = (source) =>
-	code_red.parse(source, {
+	babel.parse(source, {
 		sourceType: 'module',
-		ecmaVersion: 13,
-		locations: true
+		plugins: ['estree']
 	});
 
 /**
