diff --git a/src/compiler/interfaces.d.ts b/src/compiler/interfaces.d.ts
index 4d72cd309a1c1877d89f4fc70dbf7e74a75f7da8..dd458eebe190f508a0bf5077487ac636ecf026d5 100644
--- a/src/compiler/interfaces.d.ts
+++ b/src/compiler/interfaces.d.ts
@@ -1,5 +1,6 @@
 import { AssignmentExpression, Node, Program } from 'estree';
 import { SourceMap } from 'magic-string';
+import type { File } from '@babel/types'
 
 interface BaseNode {
 	start: number;
@@ -35,11 +36,24 @@ export interface ConstTag extends BaseNode {
 	expression: AssignmentExpression;
 }
 
-interface DebugTag extends BaseNode {
+export interface DebugTag extends BaseNode {
 	type: 'DebugTag';
 	identifiers: Node[];
 }
 
+export interface IfBlock extends BaseNode {
+  type: 'IfBlock';
+  elseif?: boolean;
+  expression: Node;
+  children: TemplateNode[];
+  else?: ElseBlock;
+}
+
+export interface ElseBlock extends BaseNode {
+  type: 'ElseBlock';
+  children: TemplateNode[];
+}
+
 export type DirectiveType =
 	| 'Action'
 	| 'Animation'
@@ -90,6 +104,11 @@ export interface SpreadAttribute extends BaseNode {
 	expression: Node;
 }
 
+export interface ShorthandAttribute extends BaseNode {
+	type: 'AttributeShorthand';
+	expression: Node;
+}
+
 export interface Transition extends BaseExpressionDirective {
 	type: 'Transition';
 	intro: boolean;
@@ -103,10 +122,13 @@ export type TemplateNode =
 	| ConstTag
 	| DebugTag
 	| MustacheTag
+  | IfBlock
+  | ElseBlock
 	| BaseNode
 	| Element
 	| Attribute
 	| SpreadAttribute
+	| ShorthandAttribute
 	| Directive
 	| Transition
 	| Comment;
@@ -127,7 +149,8 @@ export interface Parser {
 export interface Script extends BaseNode {
 	type: 'Script';
 	context: string;
-	content: Program;
+	attributes: any[];
+	content: File;
 }
 
 export interface Style extends BaseNode {
diff --git a/src/compiler/parse/acorn.js b/src/compiler/parse/acorn.js
index 85d219d8c9a024e0cb35e9554d8fef6c0458c3fd..2a104636d9801312f0099b3238e480976befe609 100644
--- a/src/compiler/parse/acorn.js
+++ b/src/compiler/parse/acorn.js
@@ -1,13 +1,13 @@
 import * as code_red from 'code-red';
+import babel from '@babel/parser';
 
 /**
  * @param {string} source
  */
 export const parse = (source) =>
-	code_red.parse(source, {
+	babel.parse(source, {
 		sourceType: 'module',
-		ecmaVersion: 13,
-		locations: true
+		plugins: ['estree']
 	});
 
 /**
diff --git a/src/compiler/parse/read/script.js b/src/compiler/parse/read/script.js
index 3d4ab358df18d42b7af4288ff3b4718d1774ec44..fcdfbd2bcd078253c671da3d01e00278c206364e 100644
--- a/src/compiler/parse/read/script.js
+++ b/src/compiler/parse/read/script.js
@@ -55,6 +55,7 @@ export default function read_script(parser, start, attributes) {
 		type: 'Script',
 		start,
 		end: parser.index,
+		attributes,
 		context: get_context(parser, attributes, start),
 		content: ast
 	};
diff --git a/types/index.d.ts b/types/index.d.ts
index 2055385b0a8af122d4cff8dcccdf456ef0b3c239..9ce00c43ccd5aec73e852622623e27a9a44e6006 100644
--- a/types/index.d.ts
+++ b/types/index.d.ts
@@ -90,7 +90,7 @@ declare module 'svelte' {
 	/**
 	 * Base class for Svelte components. Used when dev=false.
 	 *
-	 * 
+	 *
 	 */
 	class SvelteComponent_1<Props extends Record<string, any> = any, Events extends Record<string, any> = any> {
 		/**
@@ -107,11 +107,11 @@ declare module 'svelte' {
 		 *
 		 * */
 		$$set: any;
-		
+
 		$destroy(): void;
-		
+
 		$on<K extends Extract<keyof Events, string>>(type: K, callback: ((e: Events[K]) => void) | null | undefined): () => void;
-		
+
 		$set(props: Partial<Props>): void;
 	}
 	/**
@@ -137,10 +137,10 @@ declare module 'svelte' {
 	 * </script>
 	 * <MyComponent foo={'bar'} />
 	 * ```
-	 * 
+	 *
 	 */
 	export class SvelteComponent<Props extends Record<string, any> = any, Events extends Record<string, any> = any, Slots extends Record<string, any> = any> extends SvelteComponent_1<Props, Events> { [prop: string]: any;
-		
+
 		constructor(options: ComponentConstructorOptions<Props>);
 		/**
 		 * For type checking capabilities only.
@@ -163,14 +163,14 @@ declare module 'svelte' {
 		 *
 		 * */
 		$$slot_def: Slots;
-		
+
 		$capture_state(): void;
-		
+
 		$inject_state(): void;
 	}
 	/**
 	 * @deprecated Use `SvelteComponent` instead. See PR for more information: https://github.com/sveltejs/svelte/pull/8512
-	 * 
+	 *
 	 */
 	export class SvelteComponentTyped<Props extends Record<string, any> = any, Events extends Record<string, any> = any, Slots extends Record<string, any> = any> extends SvelteComponent<Props, Events, Slots> {
 	}
@@ -275,6 +275,7 @@ declare module 'svelte' {
 declare module 'svelte/compiler' {
 	import type { AssignmentExpression, Node, Program } from 'estree';
 	import type { SourceMap } from 'magic-string';
+	import type { File } from '@babel/types';
 	export { walk } from 'estree-walker';
 	interface BaseNode {
 		start: number;
@@ -287,6 +288,7 @@ declare module 'svelte/compiler' {
 	interface Text extends BaseNode {
 		type: 'Text';
 		data: string;
+		raw: string;
 	}
 
 	interface MustacheTag extends BaseNode {
@@ -310,6 +312,19 @@ declare module 'svelte/compiler' {
 		identifiers: Node[];
 	}
 
+  interface IfBlock extends BaseNode {
+    type: 'IfBlock';
+    elseif?: boolean;
+    expression: Node;
+    children: TemplateNode[];
+    else?: ElseBlock;
+  }
+
+  interface ElseBlock extends BaseNode {
+    type: 'ElseBlock';
+    children: TemplateNode[];
+  }
+
 	type DirectiveType =
 		| 'Action'
 		| 'Animation'
@@ -360,6 +375,11 @@ declare module 'svelte/compiler' {
 		expression: Node;
 	}
 
+	interface ShorthandAttribute extends BaseNode {
+		type: 'AttributeShorthand';
+		expression: Node;
+	}
+
 	interface Transition extends BaseExpressionDirective {
 		type: 'Transition';
 		intro: boolean;
@@ -373,10 +393,13 @@ declare module 'svelte/compiler' {
 		| ConstTag
 		| DebugTag
 		| MustacheTag
+    | IfBlock
+    | ElseBlock
 		| BaseNode
 		| Element
 		| Attribute
 		| SpreadAttribute
+		| ShorthandAttribute
 		| Directive
 		| Transition
 		| Comment;
@@ -384,7 +407,8 @@ declare module 'svelte/compiler' {
 	interface Script extends BaseNode {
 		type: 'Script';
 		context: string;
-		content: Program;
+		attributes: any[];
+		content: File;
 	}
 
 	interface Style extends BaseNode {
@@ -866,6 +890,7 @@ declare module 'svelte/types/compiler/preprocess' {
 
 declare module 'svelte/types/compiler/interfaces' {
 	import type { AssignmentExpression, Node, Program } from 'estree';
+	import type { File } from '@babel/types';
 	import type { SourceMap } from 'magic-string';
 	interface BaseNode {
 		start: number;
@@ -883,6 +908,7 @@ declare module 'svelte/types/compiler/interfaces' {
 	export interface Text extends BaseNode {
 		type: 'Text';
 		data: string;
+		raw: string;
 	}
 
 	export interface MustacheTag extends BaseNode {
@@ -901,11 +927,24 @@ declare module 'svelte/types/compiler/interfaces' {
 		expression: AssignmentExpression;
 	}
 
-	interface DebugTag extends BaseNode {
+	export interface DebugTag extends BaseNode {
 		type: 'DebugTag';
 		identifiers: Node[];
 	}
 
+  export interface IfBlock extends BaseNode {
+    type: 'IfBlock';
+    elseif?: boolean;
+    expression: Node;
+    children: TemplateNode[];
+    else?: ElseBlock;
+  }
+
+  export interface ElseBlock extends BaseNode {
+    type: 'ElseBlock';
+    children: TemplateNode[];
+  }
+
 	export type DirectiveType =
 		| 'Action'
 		| 'Animation'
@@ -956,6 +995,11 @@ declare module 'svelte/types/compiler/interfaces' {
 		expression: Node;
 	}
 
+	export interface ShorthandAttribute extends BaseNode {
+		type: 'AttributeShorthand';
+		expression: Node;
+	}
+
 	export interface Transition extends BaseExpressionDirective {
 		type: 'Transition';
 		intro: boolean;
@@ -969,10 +1013,13 @@ declare module 'svelte/types/compiler/interfaces' {
 		| ConstTag
 		| DebugTag
 		| MustacheTag
+    | IfBlock
+    | ElseBlock
 		| BaseNode
 		| Element
 		| Attribute
 		| SpreadAttribute
+		| ShorthandAttribute
 		| Directive
 		| Transition
 		| Comment;
@@ -993,7 +1040,8 @@ declare module 'svelte/types/compiler/interfaces' {
 	export interface Script extends BaseNode {
 		type: 'Script';
 		context: string;
-		content: Program;
+		attributes: any[];
+		content: File;
 	}
 
 	export interface Style extends BaseNode {
